kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/FilterHandler.java

```diff
@@ -298,16 +299,10 @@ private <F extends FilterResult> F handleFilteringException(Throwable t, Decoded
 
     private <F extends FilterResult> CompletableFuture<F> handleDeferredStage(DecodedFrame<?, ?> decodedFrame, CompletableFuture<F> future) {
         inboundChannel.config().setAutoRead(false);
-        var timeoutFuture = ctx.executor().schedule(() -> {
-            if (LOGGER.isWarnEnabled()) {
-                LOGGER.warn("{}: Filter {} was timed-out whilst processing {} {}", channelDescriptor(), filterDescriptor(),
-                        decodedFrame instanceof DecodedRequestFrame ? "request" : "response", decodedFrame.apiKey());
-            }
-            future.completeExceptionally(new TimeoutException("Filter %s was timed-out.".formatted(filterDescriptor())));
-        }, timeoutMs, TimeUnit.MILLISECONDS);
-        return future.whenComplete((f, throwable) -> {
-            timeoutFuture.cancel(false);
-        }).thenApplyAsync(filterResult -> filterResult, ctx.executor());
+        promiseFactory.wrapWithTimeLimit(future,
+                () -> "Filter %s was timed-out whilst processing %s %s".formatted(filterDescriptor(),
```

Comment:
What does it mean to say "a filter has been timed out"? Surely it's some request of the runtime made by the filter which has timed out?

Should we include the timeout in the message: It allows the user to more easily assess the reasonableness of the (configured or hard coded) timeout in their environment and circumstances (e.g. during some networking issue).

Reply:
Specifically its some form of deferred work that has timed out. The filter has returned an incomplete future and we've given that deferred future a timeout. Not exactly sure how to word the in a short fashion. I've taken a stab at doing so in b97f8412312a81e53ccaed31b3a14b05da7cf0c9